tests/system/marathon_common_tests.py

```diff
@@ -1268,3 +1271,169 @@ def test_ipv6_healthcheck(docker_ipv6_network_fixture):
         "The number of healthy tasks is {}, but {} was expected".format(app['tasksHealthy'], target_instances_count)
 
     client.remove_app(app['id'], True)
+
+
+@shakedown.dcos_1_11
+@pytest.mark.skipif("shakedown.ee_version() is None")
+@pytest.mark.usefixtures("wait_for_marathon_and_cleanup")
+def test_faultdomains_default():
+    """Tests if the applications that do not have a fault domain specified are launched on the same
+       fault domain as the master.
+    """
+
+    # Discover which is the "local" cluster domain
+    local_domain = get_cluster_local_domain()
+
+    # Get how many agents do we have in the local domain (zone and region) and try to start
+    # that many applications. Mesos should round-robbin the instances in the region
+    region_agents = count_agents_in_faultdomains(regions=local_domain.region)
+
+    # Create an app with default configuration
+    app_def = apps.faultdomain_app(instances=region_agents, suffix="-defaults")
+    client = marathon.create_client()
+    client.add_app(app_def)
+
+    # Wait for the app to be deployed
+    common.deployment_wait(app_id=app_def['id'])
+
+    # Ensure that all instances are running in the same region (it's ok if they belong to different zones)
+    app = client.get_app(app_def['id'])
+    common.assert_app_in_all_domains(app, regions=(local_domain.region,))
```

Comment:
As mentioned above. What are we asserting here? If I understand correctly you want the app only running in the defined region and all zones? What's the zone/region hierarchy again?

Reply:
So, we have:
* `Regions` that contain ...
* `Zones` and ...
* The tuple (region, zone) is called  `Fault Domain`

I separated the checks in region / zone. This might make it a bit more clear.